### ISSUE TITLE
A schema for tree-like structure fails with duplicate fields

### ISSUE THREAD

**Issue by lilianaziolek** [neo4j-graphql/neo4j-graphql-java#45]
My graph has a tree-like structure, where I have Category nodes, they (optionally) point to their parent category via a relation PARENT_CATEGORY. I am trying to create a GraphQL schema that would allow me to take out a category and its parent and immediate children (if any) and this fails with Schema Problem:

```
SchemaProblem{errors=[The type 'Mutation' [@-1:-1] has declared a field with a non unique name 'addCategoryParentCategory', The type 'Mutation' [@-1:-1] has declared a field with a non unique name 'deleteCategoryParentCategory']}
```

The schema looks like so:
```graphql
type Category {
    url: ID!
    name: String
    order: Int
    parentCategory: Category @relation(name: "PARENT_CATEGORY", direction: OUT)
    subCategories: [Category] @relation(name: "PARENT_CATEGORY", direction: IN)
}
```
Is it possible to get this structure to work in GraphQL? Is there a workaround for this issue, or would it need to be fixed in neo4j-graphql-java?

**Comment by jexp** [comment]
it should work. thanks for raising the issue.

it's probably a bug/oversight in the mutation generation.

**Comment by lilianaziolek** [comment]
As a side note - it does work where you have a separate type for Relation (that is what I submitted the test for in PR). I just didn't need this scenario back then, and now that I do, I discovered this does not work :)

**Comment by jexp** [comment]
ya, the mutation generation is also more recent than your contribution back then. 
it's obviously a case we've overlooked, sorry for that.

**Comment by lilianaziolek** [comment]
I feel like perhaps I should just give you my schema file so that you can bash it on some real-world scenarios 😄

**Comment by jexp** [comment]
That would definitely help. :)

I need to add the ability to embed a schema into a TCK file then it can be self-contained.

**Comment by Andy2003** [comment]
resolved by #107 (#98)
